improver/wxcode/weather_symbols.py

```diff
@@ -483,12 +396,12 @@ def _constraint_string(diagnostic, threshold_name, threshold_val):
                 )
             threshold_val = threshold.points.item()
             constraints.append(
-                _constraint_string(diagnostic, threshold_coord_name, threshold_val)
+                _get_constraint(diagnostic, threshold_coord_name, threshold_val)
             )
         if len(constraints) > 1:
             return constraints
 
-        # otherwise, return a string
+        # otherwise, return a constraint
         return constraints[0]
```

Comment:
I am looking at where this is called, and I don't think it is EVER called with `diagnostics` or `thresholds` as lists.  So I wonder if you could experiment with replacing the whole loop here with a single call?  Unless you can see a good reason why that wouldn't work?

Reply:
Yes, looks like the list version is not used. Removed the unused code.